src/extensions/extensionsHelper.ts

```
@@ -503,52 +531,284 @@ export async function publishExtensionVersionFromLocalSource(args: {
+      );
+    } else if (semver.eq(extensionSpec.version, args.latestVersion)) {
+      throw new FirebaseError(
+        `The version you are trying to publish (${clc.bold(
+          extensionSpec.version
+        )}) already exists for Extension '${clc.bold(
+          extensionRef
+        )}'. Please increment the version inside of extension.yaml.\n`,
+        { exit: 103 }
+      );
+    }
+  }
+
+  return { extensionSpec, notes };
+}
+
+/**
+ * Publishes an Extension version from a remote repo.
+ *
+ * @param publisherId the ID of the Publisher this Extension will be published under
+ * @param extensionId the ID of the Extension to be published
+ * @param repoUri the URI of the repo where this Extension's source exists
+ * @param sourceRef the commit hash, branch, or tag name in the repo to publish from
+ * @param extensionRoot the root directory that contains this Extension's source
+ * @param stage the release stage to publish
+ * @param nonInteractive whether to display prompts
+ * @param force whether to force confirmations
+ */
+export async function publishExtensionVersionFromRemoteRepo(args: {
+  publisherId: string;
+  extensionId: string;
+  repoUri: string;
+  sourceRef: string;
+  extensionRoot: string;
+  stage: ReleaseStage;
+  nonInteractive: boolean;
+  force: boolean;
+}): Promise<ExtensionVersion | undefined> {
+  const extensionRef = `${args.publisherId}/${args.extensionId}`;
+
+  // Check if Extension already exists.
+  let extension: Extension | undefined;
+  try {
+    extension = await getExtension(extensionRef);
+  } catch (err: any) {
+    // Silently fail and continue the publish flow if Extension not found.
+  }
+
+  // Prompt for repo URI and validate that it hasn't changed if previously set.
+  if (args.repoUri && !repoRegex.test(args.repoUri)) {
+    throw new FirebaseError("Repo URI follows this format: https://github.com/<user>/<repo>");
+  }
+  let repoUri = args.repoUri || extension?.repoUri;
+  if (!repoUri) {
```

This looks like it will prompt even if noninteractive=true. It should error out if repoUri is undefined and noninteractive is true